pkg/apis/pipeline/v1/pipeline_validation.go

```diff
@@ -289,15 +289,40 @@ func (pt *PipelineTask) validateWorkspaces(workspaceNames sets.String) (errs *ap
 	return errs
 }
 
-// validateRefOrSpec validates at least one of taskRef or taskSpec is specified
-func (pt PipelineTask) validateRefOrSpec() (errs *apis.FieldError) {
-	// can't have both taskRef and taskSpec at the same time
-	if pt.TaskRef != nil && pt.TaskSpec != nil {
-		errs = errs.Also(apis.ErrMultipleOneOf("taskRef", "taskSpec"))
+// validateRefOrSpec validates at least one of taskRef or taskSpec or pipelineRef or pipelineSpec is specified
+func (pt PipelineTask) validateRefOrSpec(ctx context.Context) (errs *apis.FieldError) {
+	// If alpha API's are enabled check for TaskRef or TaskSpec or PipelineRef or PipelineSpec where PipelineRef and PipelineSpec are alpha fields
+	// All non-alpha API's, check for TaskRef or TaskSpec is present
+	if pt.TaskRef == nil && pt.TaskSpec == nil && pt.PipelineRef == nil && pt.PipelineSpec == nil {
+		cfg := config.FromContextOrDefaults(ctx)
+		// if alpha API's are enabled, we
+		if cfg.FeatureFlags.EnableAPIFields == config.AlphaAPIFields {
```

Comment:
This validation seems like it may be redundant with the validation below. Maybe a cleaner way would be to add another clause to the if statement on line 324:
```
if len(nonNilFields) > 1 {
	errs = errs.Also(apis.ErrGeneric("expected exactly one, got multiple", nonNilFields...))
} else if len(nonNilFields) == 0 {
	errs = errs.Also(apis.ErrMissingOneOf("taskRef", "taskSpec", "pipelineRef", "pipelineSpec"))
}

Reply:
sure, I have moved this block of code under `else if len(nonNilFields) == 0`, thanks!